server/sonar-db-dao/src/test/java/org/sonar/db/component/ComponentDbTester.java

```diff
@@ -59,53 +62,75 @@ private SnapshotDto insertComponentAndSnapshot(ComponentDto component) {
+  }
+
+  @SafeVarargs
+  public final ComponentDto insertPublicProject(Consumer<ComponentDto>... dtoPopulators) {
+    return insertComponentImpl(newPublicProjectDto(db.getDefaultOrganization()), false, dtoPopulators);
   }
 
   @SafeVarargs
-  public final ComponentDto insertProject(Consumer<ComponentDto>... dtoPopulators) {
-    return insertComponentImpl(newProjectDto(db.getDefaultOrganization()), dtoPopulators);
+  public final ComponentDto insertPrivateProject(OrganizationDto organizationDto, Consumer<ComponentDto>... dtoPopulators) {
+    return insertComponentImpl(ComponentTesting.newPrivateProjectDto(organizationDto), true, dtoPopulators);
   }
 
   @SafeVarargs
-  public final ComponentDto insertProject(OrganizationDto organizationDto, Consumer<ComponentDto>... dtoPopulators) {
-    return insertComponentImpl(newProjectDto(organizationDto), dtoPopulators);
+  public final ComponentDto insertPublicProject(OrganizationDto organizationDto, Consumer<ComponentDto>... dtoPopulators) {
+    return insertComponentImpl(newPublicProjectDto(organizationDto), true, dtoPopulators);
+  }
+
+  public ComponentDto insertPrivateProject(OrganizationDto organizationDto) {
+    return insertComponentImpl(ComponentTesting.newPrivateProjectDto(organizationDto), true, noExtraConfiguration());
+  }
+
+  public ComponentDto insertPublicProject(OrganizationDto organizationDto) {
+    return insertComponentImpl(newPublicProjectDto(organizationDto), false, noExtraConfiguration());
   }
 
-  public ComponentDto insertProject(OrganizationDto organizationDto) {
-    return insertComponentImpl(newProjectDto(organizationDto), noExtraConfiguration());
+  public ComponentDto insertPrivateProject(OrganizationDto organizationDto, String uuid) {
+    return insertComponentImpl(newPrivateProjectDto(organizationDto, uuid), true, noExtraConfiguration());
   }
 
-  public ComponentDto insertProject(OrganizationDto organizationDto, String uuid) {
-    return insertComponentImpl(newProjectDto(organizationDto, uuid), noExtraConfiguration());
+  public ComponentDto insertPublicProject(OrganizationDto organizationDto, String uuid) {
+    return insertComponentImpl(newPublicProjectDto(organizationDto, uuid), false, noExtraConfiguration());
   }
 
   public ComponentDto insertView() {
-    return insertComponentImpl(newView(db.getDefaultOrganization()), noExtraConfiguration());
+    return insertComponentImpl(newView(db.getDefaultOrganization()), false, noExtraConfiguration());
   }
 
   public ComponentDto insertView(Consumer<ComponentDto> dtoPopulator) {
-    return insertComponentImpl(newView(db.getDefaultOrganization()), dtoPopulator);
+    return insertComponentImpl(newView(db.getDefaultOrganization()), false, dtoPopulator);
   }
 
   public ComponentDto insertView(OrganizationDto organizationDto) {
-    return insertComponentImpl(newView(organizationDto), noExtraConfiguration());
+    return insertComponentImpl(newView(organizationDto), false, noExtraConfiguration());
```

Comment:
I think we should use a varargs consumer, for to reduce the number of methods. `noExtraConfiguration` will not be needed anymore.